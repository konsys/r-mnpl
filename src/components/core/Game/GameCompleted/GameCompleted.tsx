import React from "react";
import { RoomPlayer } from "stores/Game/Rooms/RoomsModel";
import ToMainPage from "components/views/Errors/ToMainPage";

// TODO add winner page template
export default function GameCompleted({
  winner,
}: {
  winner: RoomPlayer | undefined;
}) {
  return (
    <>
      <div className="_winner">Победитель {winner && winner.name}</div>
      <ToMainPage />
    </>
  );
}
