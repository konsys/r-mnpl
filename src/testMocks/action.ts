import { ICurrentAction } from "stores/Board/ActionStore";
import {
  BoardAction,
  IncomeMessageType,
  OutcomeMessageType,
} from "types/types";

export const testDoNothingAction: ICurrentAction = {
  actionId: "testActionId",
  event: {
    action: {
      _id: "testID",
      isModal: false,
      userId: 1,
      type: IncomeMessageType.DO_NOTHING,
    },
  },
};

export const testRollDicesModal: BoardAction = {
  _id: "testBoardAction",
  type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,

  userId: 1,
  title: "testTitleRollDices",
  text: "testTextRollDices",
  isModal: true,
};
export const testRollDicesClicked: BoardAction = {
  _id: "testBoardAction",
  type: OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED,
  userId: 1,
  isModal: false,
};
