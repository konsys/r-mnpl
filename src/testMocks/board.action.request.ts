import { IBoardActionRequest, OutcomeMessageType } from "types/types";

export const testBoardActionRequest:IBoardActionRequest = {
    gameId:'testGameId',
    action:OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED
}