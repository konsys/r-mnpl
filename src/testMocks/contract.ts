import { IContract } from "types/types";

export const testContract: IContract = {
  fieldFromPrice: 1000,
  fieldIdsFrom: [1, 2],
  fieldIdsTo: [3],
  fieldToPrice: 1000,
  fromUserId: 1,
  moneyFrom: 1000,
  moneyTo: 1000,
  toUserId: 2,
};
