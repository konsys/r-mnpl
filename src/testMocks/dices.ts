import { IDices, IncomeMessageType } from "types/types";

export const testDices: IDices = {
    _id: "testId",
    isModal: true,
    type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,
    dices: [1, 2, 3],
    dicesSum: 6,
    userId: 1,
  };
  