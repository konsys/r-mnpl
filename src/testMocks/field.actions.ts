import { testField, testMortgagedField, testOwnedField } from "./field";

import { IFieldModal } from "components/views/BoardViews/Field/FieldActions/FieldActions";

export const testFieldActions: IFieldModal = {
  ...testField,
  isActive: true,

  position: { top: 20, left: 20 },
};

export const testOwnedFieldActions: IFieldModal = {
  ...testOwnedField,
  isActive: true,

  position: { top: 20, left: 20 },
};

export const testStaredFieldActions: IFieldModal = {
  ...testOwnedField,
  isActive: true,

  position: { top: 20, left: 20 },
};

export const testMortgagedFieldActions: IFieldModal = {
  ...testMortgagedField,
  isActive: true,

  position: { top: 20, left: 20 },
};
