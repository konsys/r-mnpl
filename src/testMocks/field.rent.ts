import { IFieldRent } from "types/types";

export const testRent: IFieldRent = {
  baseRent: 100,
  monopolyRent: 150,
  oneStar: 300,
  twoStar: 900,
  freeStar: 2000,
  fourStar: 2500,
  bigStar: 4000,
  paymentMultiplier: 2,
};
