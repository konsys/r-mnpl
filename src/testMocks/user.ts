import { IPlayer, IUser } from "types/types";
import { PlayerRoomStatus, RoomPlayer } from "stores/Game/Rooms/RoomsModel";

export const testPlayer1: IPlayer = {
  frags: "",
  userId: 1,
  money: 15000,
  vip: true,
  registrationType: "none",
  name: "IVAN",
  team: undefined,
  avatar: "testAvatar1",
  createdAt: new Date(),
  updatedAt: new Date(),
  isActive: false,
  isBlocked: true,
  isActing: true,
  gameId: "",
  doublesRolledAsCombo: 0,
  jailed: 0,
  unjailAttempts: 0,
  meanPosition: 0,
  creditPayRound: false,
  creditNextTakeRound: 0,
  score: 0,
  additionalTime: 0,
  timeReduceLevel: 0,
  creditToPay: 0,
  canUseCredit: false,
  moveOrder: 0,
};

export const testPlayer2: IPlayer = {
  frags: "",
  userId: 1,
  money: 15000,
  vip: true,
  registrationType: "none",
  name: "DIMAN",
  team: undefined,
  avatar: "testAvatar2",
  createdAt: new Date(),
  updatedAt: new Date(),
  isActive: false,
  isBlocked: true,
  isActing: false,
  gameId: "",
  doublesRolledAsCombo: 0,
  jailed: 0,
  unjailAttempts: 0,
  meanPosition: 0,
  creditPayRound: false,
  creditNextTakeRound: 0,
  score: 0,
  additionalTime: 0,
  timeReduceLevel: 0,
  creditToPay: 0,
  canUseCredit: false,
  moveOrder: 0,
};

export const testUser: IUser = {
  userId: 1,
  vip: false,
  name: "IVAN",
  isActive: true,
  isBlocked: false,
};
export const test5User: IUser = {
  userId: 5,
  vip: false,
  name: "NIKOLA",
  isActive: true,
  isBlocked: false,
};
export const testAvatarUser: IUser = {
  avatar: "/test/avatar.jpg",
  userId: 2,
  vip: false,
  name: "IVAN",
  isActive: true,
  isBlocked: false,
};
export const testVipUser: IUser = {
  userId: 3,
  vip: true,
  name: "OLEG",
  isActive: true,
  isBlocked: false,
};

export const testRoomUser: RoomPlayer = {
  playerRoomStatus: PlayerRoomStatus.ACITVE,
  userId: 4,
  vip: true,
  name: "MISHA",
  isActive: true,
  isBlocked: false,
};
