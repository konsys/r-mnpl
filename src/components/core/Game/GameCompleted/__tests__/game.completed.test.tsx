import GameCompleted from "../GameCompleted";
import React from "react";
import { shallow } from "enzyme";
import { testRoomUser } from "testMocks/user";

describe("Buy gallery test", () => {
  it("should render undefined", () => {
    expect(shallow(<GameCompleted winner={undefined} />)).toMatchSnapshot();
  });
  it("should render winner", () => {
    expect(shallow(<GameCompleted winner={testRoomUser} />)).toMatchSnapshot();
  });

  it("should render winner name", () => {
    expect(
      shallow(<GameCompleted winner={testRoomUser} />)
        .find("._winner")
        .text()
    ).toBe(`Победитель ${testRoomUser.name}`);
  });
});
