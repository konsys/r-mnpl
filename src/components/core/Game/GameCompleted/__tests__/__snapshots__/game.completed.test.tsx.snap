// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buy gallery test should render undefined 1`] = `ShallowWrapper {}`;

exports[`Buy gallery test should render winner 1`] = `ShallowWrapper {}`;
